waspc/src/Wasp/Project/Analyze.hs

```diff
@@ -86,12 +97,26 @@ constructAppSpec waspDir options decls = do
     [] -> Right appSpec
     validationErrors -> Left $ map show validationErrors
 
-findWaspFile :: Path' Abs (Dir WaspProjectDir) -> IO (Either String (Path' Abs File'))
+findWaspFile :: Path' Abs (Dir WaspProjectDir) -> IO (Maybe (Path' Abs File'))
 findWaspFile waspDir = do
   files <- fst <$> IOUtil.listDirectory waspDir
-  return $ maybeToEither "Couldn't find a single *.wasp file." $ (waspDir </>) <$> find isWaspFile files
+  return $ (waspDir </>) <$> find isWaspFile files
```

Comment:
Why did you decide to take this error message out and then apply it out there? It is not terrible, but it seemed to me that it makes more sense for it to be generated here, where there is more context on what happened.

Reply:
I wanted to do it for a long time now, when you suggested this: https://github.com/wasp-lang/wasp/pull/753#discussion_r1004229754.

You might have meant something else, but `findWaspFile :: Path' Abs (Dir WaspProjectDir) -> IO (Maybe (Path' Abs File'))` seems like the cleanest name/type combo.

Reply:
Yeah that is what I meant, damn me from past :D. But I was right I guess. Both approaches make sense in a way to me now. Ok, all good.